packages/back-end/src/app.ts

```diff
@@ -197,8 +197,25 @@ app.post(
   slackController.postIdeas
 );
 
-// increase max payload json size to 1mb
-app.use(bodyParser.json({ limit: "1mb" }));
+// Accept cross-origin requests from the frontend app
+const origins: (string | RegExp)[] = [APP_ORIGIN];
+if (CORS_ORIGIN_REGEX) {
+  origins.push(CORS_ORIGIN_REGEX);
+}
+
+// For non-api routes, set the global cors
+app.use(
+  cors({
+    credentials: true,
+    origin: origins,
+  })
+);
+
+// increase default max payload json size to 1mb and allow for per-route specific payload sizes
+app.use((req, res, next) => {
+  // Can add conditional logic on req.path and req.method to set the body limit here if needed
+  return bodyParser.json({ limit: "1mb" })(req, res, next);
+});
```

Comment:
The ordering here is important as we need the cors definition to be higher up in the chain and let it be overridden later. 

On the other hand, bodyParser limits can't be overridden, so I added this section for conditional logic. We ended up deciding to use the same 1mb limit for saved group payloads so this isn't actually necessary right now, but it might save some time in the future if we decide to expand the limit on any particular endpoint.

Reply:
Have you tested that overriding cors for individual routes works as expected?  Specifically thinking of the routes which only specify origin and not credentials:

```ts
  cors({
    origin: "*",
  }),
```

Reply:
I'm fairly sure I did but it's been a month and looking back at these changes I'm thinking it'd be safer to not add this logic now. I've at least gone through the exercise of figuring out how to change the limit, so that should save us time if we need to in the future.